zerver/lib/actions.py

```diff
@@ -4865,9 +4865,12 @@ def do_send_user_group_update_event(user_group: UserGroup, data: Dict[str, Any])
     send_event(event, active_user_ids(user_group.realm_id))
 
 def do_update_user_group_name(user_group: UserGroup, name: str) -> None:
-    user_group.name = name
-    user_group.save(update_fields=['name'])
-    do_send_user_group_update_event(user_group, dict(name=name))
+    try:
+        user_group.name = name
+        user_group.save(update_fields=['name'])
+        do_send_user_group_update_event(user_group, dict(name=name))
```

Comment:
This last line doesn't need to be inside the `try` block; I moved it out for clarity.

(It's good practice to keep `try/except` blocks as small as possible, to make clear which code you're doing special error handling for)